django/contrib/auth/forms.py

```diff
@@ -54,11 +54,17 @@ def get_context(self, name, value, attrs):
                 for key, value_ in hasher.safe_summary(value).items():
                     summary.append({"label": gettext(key), "value": value_})
         context["summary"] = summary
+        context["usable_password"] = len(summary) > 1
```

Comment:
I believe that is cleaner to define using `value.startswith(UNUSABLE_PASSWORD_PREFIX)`. So, something like:
```python
        usable_password = not value.startswith(UNUSABLE_PASSWORD_PREFIX)
        if not value or not usable_password:
            summary.append({"label": gettext("No password set.")})
        else:
            # ...
        context["usable_password"] = usable_password
```

Reply:
On a second thought, since this is the Widget class, shall we instead add to the context the label and action of the button? My initial thinking is that approach would be cleaner and simpler in the template. What do you think?

Reply:
@nessita What's the right way to get the admin namespace from within the widget? I would need it to reverse the set password URL. The template uses a `../bla` which indeed is a bit hacky. The reason might be that the admin namespace is not readily available.

Admin forms get the admin site passed to their `__init__` function. But that pattern would require to change the signature of the `UserChangeForm` which I would like to avoid. 

I have moved the button label from the template to the `get_context` method, however.